File: MyAppVolley/app/src/main/java/com/example/myappvolley/UserListAdapter.java
package com.example.myappvolley;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class UserListAdapter extends BaseAdapter {

    private Context context;
    private ArrayList<UserModel> dataModelArrayList;

    public UserListAdapter(Context context, ArrayList<UserModel> dataModelArrayList) {

        this.context = context;
        this.dataModelArrayList = dataModelArrayList;
    }

    @Override
    public int getViewTypeCount() {
        return getCount();
    }
    @Override
    public int getItemViewType(int position) {

        return position;
    }

    @Override
    public int getCount() {
        return dataModelArrayList.size();
    }

    @Override
    public Object getItem(int position) {
        return dataModelArrayList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;

        if (convertView == null) {
            holder = new ViewHolder();
            LayoutInflater inflater = (LayoutInflater) context
                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.lv_player, null, true);


            holder.id = (TextView) convertView.findViewById(R.id.id);
            holder.nombre = (TextView) convertView.findViewById(R.id.nombre);
            holder.estado = (TextView) convertView.findViewById(R.id.estado);

            convertView.setTag(holder);
        }else {
            // the getTag returns the viewHolder object set as a tag to the view
            holder = (ViewHolder)convertView.getTag();
        }


        holder.id.setText("id: "+dataModelArrayList.get(position).getid());
        holder.nombre.setText("nombre: "+dataModelArrayList.get(position).getNombre());
        holder.estado.setText("estado: "+dataModelArrayList.get(position).getestado());

        return convertView;
    }



    private class ViewHolder {

        protected TextView id, nombre, estado;

    }

}

File: MyAppKotlin/app/src/main/java/com/example/myappkotlin/Models/UserListAdapter.kt
package com.example.myappkotlin.Models

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.example.myappkotlin.R
import java.util.ArrayList
;

class UserListAdapter(private val context: Context, private val dataModelArrayList: ArrayList<UserModel>) :
    BaseAdapter() {

    override fun getViewTypeCount(): Int {
        return count
    }

    override fun getItemViewType(position: Int): Int {

        return position
    }

    override fun getCount(): Int {
        return dataModelArrayList.size
    }

    override fun getItem(position: Int): Any {
        return dataModelArrayList[position]
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val holder: ViewHolder

        if (convertView == null) {
            holder = ViewHolder()
            val inflater = context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            convertView = inflater.inflate(R.layout.lv_player, null, true)


            holder.id = convertView!!.findViewById(R.id.id) as TextView
            holder.nombre = convertView.findViewById(R.id.nombre) as TextView
            holder.estado = convertView.findViewById(R.id.estado) as TextView

            convertView.tag = holder
        } else {
            // the getTag returns the viewHolder object set as a tag to the view
            holder = convertView.tag as ViewHolder
        }


        holder.id!!.text = "id: " + dataModelArrayList[position].getid()
        holder.nombre!!.text = "nombre: " + dataModelArrayList[position].getNombre()
        holder.estado!!.text = "estado: " + dataModelArrayList[position].getestado()

        return convertView
    }

    private inner class ViewHolder {

        var id: TextView? = null
        var nombre: TextView? = null
        var estado: TextView? = null

    }

}